File: src/main/java/com/mercateo/jsonschema/schema/PropertyChecker.java
package com.mercateo.jsonschema.schema;

import com.mercateo.jsonschema.property.Property;

import java.util.function.Predicate;

/**
 * this class checks, if a field of a bean should be contained in the schema
 *
 * @author joerg_adler
 */
public interface PropertyChecker extends Predicate<Property> {
    static PropertyChecker fromPredicate(Predicate<Property> predicate) {
        return predicate::test;
    }
}

File: src/main/java/com/mercateo/jsonschema/mapper/PropertyChecker.kt
package com.mercateo.jsonschema.mapper

import com.mercateo.jsonschema.property.Property

import java.util.function.Predicate

interface PropertyChecker : Predicate<Property<*, *>> {

    companion object {
        fun fromPredicate(predicate: Predicate<Property<*, *>>): PropertyChecker
                = object : PropertyChecker {
            override fun test(t: Property<*, *>): Boolean {
                return predicate.test(t)
            }
        }
    }
}
